.azure-pipelines.yml

```diff
@@ -0,0 +1,206 @@
+  GO_DEP_VERSION: '0.5.4'
+  PYTHON_PIPENV_VERSION: '2018.11.26'
+  RUST_VERSION: '1.35.0'
+  STACK_VERSION: '2.1.3.20190715'
+  VIRTUALENV_VERSION: '20.0.14'
+
+stages:
+- stage: build_test
+  displayName: Build and Test
+  jobs:
+  # TODO: Consider running the Docker job only if the Dockerfile was changed.
+  - job: 'Docker'
+    pool:
+      vmImage: 'ubuntu-18.04'
+    steps:
+    - task: Docker@2
+      inputs:
+        command: 'build'
+        Dockerfile: 'Dockerfile'
+        tags: |
+          ort
+    # TODO: Add a step to run ORT in the Docker container, e.g. the requirements task, to verify that the Docker
+    #       container works.
+
+  # TODO: Add a separate job to run detekt to prevent running the matrix build if there are detekt errors?
+  #       Downside is that this requires separate cloning. Alternative is to run detekt only on one OS, or to run detekt
+  #       in a separate parallel job for early feedback, and disable detekt in the other jobs.
+  - job: 'Gradle'
+    strategy:
+      matrix:
+        Linux:
+          imageName: 'ubuntu-18.04'
+          gradleWrapper: 'gradlew'
+          FLUTTER_HOME: '/opt/flutter'
+        # TODO: Add a mac build.
+        Windows:
+          imageName: 'windows-2019'
+          gradleWrapper: 'gradlew.bat'
+          FLUTTER_HOME: 'C:\flutter'
+    pool:
+      vmImage: $(imageName)
+    steps:
+    - task: UsePythonVersion@0
+      displayName: Enable Python 3.6
+      inputs:
+        versionSpec: '3.6'
+        addToPath: true
+        architecture: 'x64'
+
+    # Linux requirements.
+    - bash: |
+        # Install apt packages.
+        sudo apt-get -qq install cvs
+
+        # Uninstall mono-devel because it contains the "lc" command which conflicts with BoyterLc.
+        sudo apt-get -qq remove mono-devel
+
+        # Install NPM packages.
+        sudo npm install -g bower@$BOWER_VERSION
+
+        # Install Python packages.
+        export PATH=$PATH:~/.local/bin
+        pip install --user \
+          conan==$CONAN_VERSION \
+          pipenv==$PYTHON_PIPENV_VERSION \
+          reuse \
+          virtualenv==$VIRTUALENV_VERSION
+        conan user # Create the conan data directory. Automatic detection of your arch, compiler, etc.
+
+        # Install Ruby packages.
+        sudo gem install bundler -v $BUNDLER_VERSION
+
+        # Downgrade Rust, because the CargoSubcrateTest fails with the pre-installed version.
+        rustup default $RUST_VERSION
+
+        # Install Flutter.
+        curl -Os https://storage.googleapis.com/flutter_infra/releases/stable/linux/flutter_linux_$FLUTTER_VERSION.tar.xz
+        tar xf flutter_linux_$FLUTTER_VERSION.tar.xz -C $(dirname $FLUTTER_HOME)
+        rm flutter_linux_$FLUTTER_VERSION.tar.xz
+        export PATH="$PATH:$FLUTTER_HOME/bin:$FLUTTER_HOME/bin/cache/dart-sdk/bin"
+        flutter config --no-analytics
+        flutter doctor
+
+        # Install git-repo.
+        mkdir ~/bin
+        export PATH=$PATH:~/bin
```

Comment:
How about aligning on using either `~/.local/bin` (see above) or `~/bin`, but not both?

Reply:
`.local/bin` is usually only used by Pip, I didn't want to mix those two and keep everything similar to the AppVeyor config, where both paths are already in the default path.

Reply:
FYI, [the ~/.local/bin directory is part of the systemd file-hierarchy spec](https://unix.stackexchange.com/a/392710/53328) (in contrast to `~/bin`, which AFAIK is not standardized) and as such in general used by more tools than just Pip. That's why I thought it makes sense to align on it.

Reply:
Ok, changed.